### DISCUSSION TITLE
Two mass system

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#2180] [Q&A]
Dear @jjonkman 
![image](https://github.com/OpenFAST/openfast/assets/107483730/17b85535-9e8b-47b4-8b8e-856cb6e70fa5)
![image](https://github.com/OpenFAST/openfast/assets/107483730/4e66dfee-5ad4-4b46-b33e-efd4af7d5c74)
I established a two mass system drivetrain dynamic model based on the equation in Figure 1. 
Where 1/J_w=1/38759228/97^2,  1/J_e=1/534.116,  K=867637000/97^2,  D=6215000/97^2 
Q1:Now I want to run this simulink model, but I don't know what the two input parameters (Tw and Te)represent? 
Q2:How should I input these two parameters?
Q3:What units were these two parameters inputted?

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

Tw and Te in your two-mass model are the aerodynamic torque and electrical generator torque, respectively.  The units should be consistent with rest of your equations of motion (presumably they should have the units of N*m).   I'm not sure why you appear to be expressing the parameters relative to the high-speed shaft; I would think it would be more naturally to express the parameters relative to the low-speed shaft.

A similar two-mass generator-rotor drivetrain model was recently discussed on our forum: https://forums.nrel.gov/t/drivetrain-dynamic-model/1864.  I suspect the discussion there is useful to you.

Best regards,
